packages/core/src/components/SupportButton/SupportButton.tsx

```diff
@@ -78,13 +76,42 @@ const SupportListItem = ({ item }: { item: SupportItem }) => {
   );
 };
 
-export const SupportButton = ({ children }: PropsWithChildren<Props>) => {
-  const { items } = useSupportConfig();
+const createDisplayedItems = (
+  showCondition: Boolean | undefined,
+  generalItems: SupportItem[],
+  supportItems: SupportItem[] | undefined,
+) => {
+  if (showCondition === undefined) return generalItems;
```

Comment:
nit: it's not common to avoid using braces around if statements in the code base. I think a ternary would probably be work just as well to make this succinct.

Also, are you trying to model a 3 state boolean with null/intermediate? This seems like it could be confusing TBH. I'd say we want a boolean, then it should just be on/off (e.g. `hideGeneralItems` or something). Otherwise a string union type is probably more clear imo.

Reply:
I tried to use a ternary operator. but the lint rules didn't allow me to commit the changes. I think I'm gonna keep the boolean flag, just only I'll change the flag name to be more specific